examples_tests/23.Autoexposure/main.cpp

```diff
@@ -4,20 +4,771 @@
+			asset::SBufferRange<video::IGPUBuffer> bufferRange = {};
+			bufferRange.buffer = toneParamsLumaOutputBuffer[acquiredNextFBO];
+			bufferRange.offset = offsetPastLumaHistory;
+			bufferRange.size = sizeof(tonemappingParams) - bufferRange.offset;
+
+			// memory dependency to ensure all the compute shader read for the tonemapping dispatch
+			// has finished before we try to transfer new stuff to the buffer
+			video::IGPUCommandBuffer::SBufferMemoryBarrier prevParamsReadBarrier = {};
+			prevParamsReadBarrier.barrier.srcAccessMask = asset::EAF_SHADER_READ_BIT;
+			prevParamsReadBarrier.barrier.dstAccessMask = asset::EAF_TRANSFER_WRITE_BIT;
+			prevParamsReadBarrier.srcQueueFamilyIndex = ~0u;
+			prevParamsReadBarrier.dstQueueFamilyIndex = ~0u;
+			prevParamsReadBarrier.buffer = bufferRange.buffer;
+			prevParamsReadBarrier.offset = bufferRange.offset;
+			prevParamsReadBarrier.size = bufferRange.size;
+
+			cb->pipelineBarrier(
+				asset::EPSF_COMPUTE_SHADER_BIT,
+				asset::EPSF_TRANSFER_BIT,
+				asset::EDF_BY_REGION_BIT,
+				0u, nullptr,
+				1u, &prevParamsReadBarrier,
+				0u, nullptr);
+
+			uint32_t waitSemCount = 0u;
+			video::IGPUSemaphore *const* waitSem = nullptr;
+			const asset::E_PIPELINE_STAGE_FLAGS* waitStages = nullptr;
+			utilities->updateBufferRangeViaStagingBuffer(
+				cb.get(),
+				nullptr,
+				graphicsComputeQueue,
+				bufferRange,
+				paramPtr + offsetPastLumaHistory,
+				waitSemCount, waitSem, waitStages);
+		}
+
+		video::IGPUCommandBuffer::SRenderpassBeginInfo beginInfo = {};
+		{
+			VkRect2D area;
+			area.offset = { 0,0 };
+			area.extent = { WIN_W, WIN_H };
+			asset::SClearValue clear[2] = {};
+			clear[0].color.float32[0] = 1.f;
+			clear[0].color.float32[1] = 0.f;
+			clear[0].color.float32[2] = 0.f;
+			clear[0].color.float32[3] = 1.f;
+			clear[1].depthStencil.depth = 0.f;
+
+			beginInfo.clearValueCount = 2u;
+			beginInfo.framebuffer = fbos[acquiredNextFBO];
+			beginInfo.renderpass = renderpass;
+			beginInfo.renderArea = area;
+			beginInfo.clearValues = clear;
+		}
+
+		cb->beginRenderPass(&beginInfo, nbl::asset::ESC_INLINE);
+		cb->bindDescriptorSets(
+			asset::EPBP_GRAPHICS,
+			fstGraphicsPipeline->getRenderpassIndependentPipeline()->getLayout(),
+			3, 1,
+			&fstDS[acquiredNextFBO].get());
+		cb->bindGraphicsPipeline(fstGraphicsPipeline.get());
+
+		ext::FullScreenTriangle::recordDrawCalls(cb.get());
+
+		cb->endRenderPass();
+
+		// transfer back to GENERAL
+		{
+			layoutTransBarrier.barrier.srcAccessMask = asset::EAF_SHADER_READ_BIT;
+			layoutTransBarrier.barrier.dstAccessMask = asset::EAF_SHADER_WRITE_BIT;
+			layoutTransBarrier.oldLayout = asset::EIL_SHADER_READ_ONLY_OPTIMAL;
+			layoutTransBarrier.newLayout = asset::EIL_GENERAL;
+		}
+		cb->pipelineBarrier(
+			asset::EPSF_FRAGMENT_SHADER_BIT,
+			asset::EPSF_COMPUTE_SHADER_BIT,
+			asset::EDF_BY_REGION_BIT,
+			0u, nullptr,
+			0u, nullptr,
+			1u, &layoutTransBarrier);
+
+		cb->end();
```

Comment:
can we not image-store directly to the swapchain image and skip any renderpasses, etc.?